codex/node.nim

```diff
@@ -60,10 +60,29 @@ proc connect*(
   addrs: seq[MultiAddress]): Future[void] =
   node.switch.connect(peerId, addrs)
 
-# TODO: move code that retrieves blocks in manifest into blockstore
-proc retrieve*(
+proc fetchBlocksJob(
   node: CodexNodeRef,
-  cid: Cid): Future[?!LPStream] {.async.} =
+  manifest: Manifest,
+  batch = FetchBatch) {.async.} =
+  ## Initiates requests to all blocks in the manifest
+  ## and stores the blocks on the local store
+  ##
+
+  try:
+    let
+      batch = max(1, manifest.blocks.len div batch)
+    trace "Prefetching in batches of", FetchBatch
+    for blks in manifest.blocks.distribute(batch, true):
+      discard await allFinished(
+        blks.mapIt( node.blockStore.getBlock( it ) ))
+  except CatchableError as exc:
+    trace "Exception prefetching blocks", exc = exc.msg
```

Comment:
The onStore callback needs to be able to see whether the fetching of the blocks failed or succeeded (see reasoning below).